migrator/src/test/java/com/google/phonenumbers/migrator/MigrationJobTest.java

```diff
@@ -133,20 +139,58 @@ public void performMultipleMigration_nextRecipeNotFound_expectException() throws
   }
 
   @Test
-  public void performMultipleMigration_expectMigration() throws IOException {
+  public void customRecipe_multipleMigration_expectMigration() throws IOException {
     String recipesPath = TEST_DATA_PATH + "testRecipesFile.csv";
     String staleNumber = "15321";
+    String countryCode = "44";
     String migratedNumber = "130211";
     MigrationJob job = MigrationFactory
-        .createMigration(staleNumber, "GB", Paths.get(recipesPath));
+        .createMigration(staleNumber, countryCode, Paths.get(recipesPath));
 
     RangeSpecification testRecipePrefix = RangeSpecification.from(DigitSequence.of("15"));
     int testRecipeLength = 5;
     RangeKey recipeKey = RangeKey.create(testRecipePrefix, Collections.singleton(testRecipeLength));
 
-    ImmutableList<MigrationResult> migratedNums = job.getMigratedNumbersForRecipe(recipeKey);
-    assertThat(migratedNums.stream().map(MigrationResult::getMigratedNumber)
+    MigrationReport migratedNums = job.getMigrationReportForRecipe(recipeKey)
+        .orElseThrow(() -> new RuntimeException("Migration was expected but from found"));
+
+    assertThat(migratedNums.getValidMigrations().stream().map(MigrationResult::getMigratedNumber)
         .collect(Collectors.toList()))
         .containsExactly(DigitSequence.of(migratedNumber));
   }
+
+  @Test
+  public void standardMigration_invalidNumberNoRecipe_expectNoMigration() throws IOException {
+    String invalidNumber = "1234567";
+    String countryCode = "44";
```

Comment:
I think a lot of these variables could be class level variables (ie. private static final COUNTRY_CODE = "44";)